src/main/java/com/uber/cadence/migration/MigrationInterceptor.java

```diff
@@ -121,4 +124,71 @@ private MigrationDecision shouldMigrate(
       SyncWorkflowDefinition workflowDefinition, WorkflowExecuteInput input) {
     return new MigrationDecision(true, "");
   }
+
+  @Override
+  public void continueAsNew(
+      Optional<String> workflowType, Optional<ContinueAsNewOptions> options, Object[] args) {
+
+    int version = getVersion(versionChangeID, Workflow.DEFAULT_VERSION, versionV1);
+    switch (version) {
+      case versionV1:
+        WorkflowInfo workflowInfo = Workflow.getWorkflowInfo();
+        WorkflowExecutionStartedEventAttributes startedEventAttributes =
+            workflowInfo.getWorkflowExecutionStartedEventAttributes();
+        if (isChildWorkflow(startedEventAttributes)) {
+          next.continueAsNew(workflowType, options, args);
+        }
+        MigrationDecision decision =
+            Workflow.sideEffect(MigrationDecision.class, () -> new MigrationDecision(true, ""));
+        if (decision.shouldMigrate) {
+          StartWorkflowExecutionRequest request =
+              new StartWorkflowExecutionRequest()
+                  .setDomain(workflowInfo.getDomain())
+                  .setWorkflowId(workflowInfo.getWorkflowId())
+                  .setTaskList(new TaskList().setName(startedEventAttributes.taskList.getName()))
+                  .setInput(startedEventAttributes.getInput())
```

Comment:
This is incorrect. We need to use serialized `args`. When a workflow continue as new, they can change the input.